addons/account/models/account_move.py

```diff
@@ -4219,10 +4241,17 @@ def write(self, vals):
         if account_to_write and account_to_write.deprecated:
             raise UserError(_('You cannot use a deprecated account.'))
 
+        inalterable_fields = set(self._get_integrity_hash_fields()).union({'inalterable_hash', 'secure_sequence_number'})
+        violated_fields = ", ".join([
+            self._fields[violated_field].string
```

Comment:
This is getting the source string, not the translated one. Have a look at `get_description` and/or `fields_get`